src/server/redis_reply.cc

```diff
@@ -20,15 +20,37 @@
 
 #include "redis_reply.h"
 
+#include <map>
 #include <numeric>
 
+const std::map<Status::Code, std::string> redisErrorPrefixMapping = {
+    {Status::RedisErrorNoPrefix, ""},         {Status::RedisNoProto, "NOPROTO"},
+    {Status::RedisLoading, "LOADING"},        {Status::RedisMasterDown, "MASTERDOWN"},
+    {Status::RedisNoScript, "NOSCRIPT"},      {Status::RedisNoAuth, "NOAUTH"},
+    {Status::RedisWrongType, "WRONGTYPE"},    {Status::RedisReadOnly, "READONLY"},
+    {Status::RedisExecAbort, "EXECABORT"},    {Status::RedisMoved, "MOVED"},
+    {Status::RedisCrossSlot, "CROSSSLOT"},    {Status::RedisTryAgain, "TRYAGAIN"},
+    {Status::RedisClusterDown, "CLUSTERDOWN"}};
+
 namespace redis {
 
 void Reply(evbuffer *output, const std::string &data) { evbuffer_add(output, data.c_str(), data.length()); }
 
 std::string SimpleString(const std::string &data) { return "+" + data + CRLF; }
 
-std::string Error(const std::string &err) { return "-" + err + CRLF; }
+std::string Error(const Status &s) { return RESP_PREFIX_ERROR + StatusToRedisError(s); }
+
+std::string StatusToRedisError(const Status &s) {
```

Comment:
```suggestion
std::string StatusToRedisErrorMsg(const Status &s) {
```
Maybe this name is better.

Reply:
Then we can distinguish between redis error message and resp errors.

Reply:
done